airflow/operators/sql.py

```diff
@@ -467,6 +467,269 @@ def push(self, meta_data):
         self.log.info("Log from %s:\n%s", self.dag_id, info)
 
 
+def _get_failed_tests(checks):
+    return [
+        f"\tCheck: {check}, " f"Check Values: {check_values}\n"
+        for check, check_values in checks.items()
+        if not check_values["success"]
+    ]
+
+
+class SQLColumnCheckOperator(BaseSQLOperator):
+    """
+    Performs one or more of the templated checks in the column_checks dictionary.
+    Checks are performed on a per-column basis specified by the column_mapping.
+
+    :param table: the table to run checks on.
+    :param column_mapping: the dictionary of columns and their associated checks, e.g.:
+    {
+        'col_name': {
+            'null_check': {
+                'equal_to': 0,
+            },
+            'min': {
+                'greater_than': 5,
+                'leq_than': 10,
+                'tolerance': 0.2,
+            },
+            'max': {
+                'less_than': 1000,
+                'geq_than': 10,
+                'tolerance': 0.01
+            }
+        }
+    }
```

Comment:
For these pesky build docs errors, you can try adding a `.. code-block::` directive and house these example dicts under them. Something like:
```rst
:param table: the table to run checks on.
:param column_mapping: the dictionary of columns and their associated checks, e.g.:

    .. code-block::

        column_mapping = {
            'col_name': {
                'null_check': {
                    'pass_value': 0,
                },
                'min': {
                    'pass_value': 5,
                    'tolerance': 0.2,
                }
            }
        }

:param conn_id: the connection ID used to connect to the database.
:param database: name of database which overwrite the defined one in connection
```
Check out the docstrings for [`chain()`](https://github.com/apache/airflow/blob/fd4e344880b505cfe53a97a6373d329515bbc7a3/airflow/models/baseoperator.py#L1520) and [`cross_downstream()`](https://github.com/apache/airflow/blob/fd4e344880b505cfe53a97a6373d329515bbc7a3/airflow/models/baseoperator.py#L1649) for examples.